src/transformers/utils/doc.py

```diff
@@ -1143,6 +1144,15 @@ def docstring_decorator(fn):
         func_doc = (fn.__doc__ or "") + "".join(docstr)
         output_doc = "" if output_type is None else _prepare_output_docstrings(output_type, config_class)
         built_doc = code_sample.format(**doc_kwargs)
+        if revision is not None:
+            if re.match(r"^refs/pr/\\d+", revision):
+                raise ValueError(
+                    f"The provided revision '{revision}' is incorrect. It should point to"
+                    " a pull request reference on the hub like 'refs/pr/6'"
+                )
+            built_doc = built_doc.replace(
+                f'from_pretrained("{checkpoint}")', f'from_pretrained("{checkpoint}", revision="{revision}")'
+            )
```

Comment:
Won't this replace for both the model and any processor classes?

Reply:
yep. In case a pr update the tokenizer / processor as well. Even if they are not updated they should still be usable!